Update releases instructions to describe order (#40)
ember-learn/handbook: ember-releases.md

- Release page (index, stable, beta, LTS)
- [Glitch Ember starter](https://emberjs.glitch.me)

> The Guides and API docs should be published just before the Release blog post. That way, links can be included in the blog post. All other steps come after the blog post.

Reviewer: I think the word "before" may be a little confusing. For clarity, what do you think of:
```suggestion
The Guides and API docs should be published first, followed by the Release blog post. This way, links to the Guides and API docs in the Release blog post. All other steps come after the blog post.
```
?
